registry/storage/blobserver.go

```diff
@@ -36,16 +36,15 @@ func (bs *blobServer) ServeBlob(ctx context.Context, w http.ResponseWriter, r *h
 
 	redirectURL, err := bs.driver.URLFor(ctx, path, map[string]interface{}{"method": r.Method})
 
-	switch err {
-	case nil:
+	if err == nil {
```

Comment:
I changed ErrUnsupportedMethod from var to type. This meant switch on err doesn't work any more.